src/util/files.c

```diff
@@ -185,20 +185,26 @@ static int remove_tree_with_ctx(TALLOC_CTX *mem_ctx,
     DIR *rootdir = NULL;
     int ret, err;
     int dir_fd;
+    int log_level;
 
     dir_fd = sss_openat_cloexec(parent_fd, dir_name,
                             O_RDONLY | O_DIRECTORY | O_NOFOLLOW, &ret);
     if (dir_fd == -1) {
         ret = errno;
-        DEBUG(SSSDBG_CRIT_FAILURE, "Cannot open %s: [%d]: %s\n",
+        if (ret == ENOENT) {
```

Comment:
`ENOENT` addresses the issue in the BZ but maybe other flags should be included here too? Like `EACCES` or `ENOTDIR` as they looks to be not critical errors too. Or we will include other errors in the list when dedicated BZ will arrive in future?

Reply:
For all other error codes level is reduced as well: from `SSSDBG_CRIT_FAILURE` to `SSSDBG_MINOR_FAILURE`. This level isn't enabled by default and doesn't trigger a backtrace.

As to "why"... It's difficult to say what error is "expected" - it's up to a function user to decide. But in general "missing" dir feels not as suspicious as "wrong access rights".